gpflow/utilities/utilities.py

```diff
@@ -30,6 +30,7 @@
 Path = str
 Accumulator = Tuple[Path, State]
 TraverseUpdateCallable = Callable[[TraverseInput, Path, State], State]
+M = TypeVar("M", bound=tf.Module)
```

Comment:
could you move this back next to the `deepcopy` definition so that the context can make clear what this is & what it's used for ?